frontend/model/contracts/identity.js

```diff
@@ -180,6 +190,10 @@ sbp('chelonia/defineContract', {
       process ({ data }, { state }) {
         // NOTE: this method is always created by another
         const { contractID } = data
+        if (!state.chatRooms) {
+          // When creating a DM, we may not have the `.chatRooms` property
+          state.chatRooms = Object.create(null)
+        }
```

Comment:
Why would we not have the `.chatRooms` property? It's created by the constructor

Reply:
That requires the constructor being called. It won't be if the decryption key isn't available (for example, because it's been rotated).

Reply:
Could you add a comment here explaining this? I think most people reading this code will have no idea about this.

Reply:
Sure, I can, but then this is a caveat that IMO should be well-documented in the Chelonia documentation. Because the constructor is a 'regular' action, it needs to be readable for it to run. Hence, it's not guaranteed that the constructor will run before other actins. Actually, because of the way things are implemented, the constructor could _never_ be run, if it's not issued (we issue `OP_CONTRACT` followed by `OP_ACTION_ENCRYPTED`; one could issue just `OP_CONTRACT`).